tests/lib/linter/linter.js

```diff
@@ -2733,14 +2733,31 @@ describe("Linter", () => {
     });
 
     describe("when evaluating code with comments which have colon in its value", () => {
-        const code = "/* eslint max-len: [2, 100, 2, {ignoreUrls: true, ignorePattern: \"data:image\\/|\\s*require\\s*\\(|^\\s*loader\\.lazy|-\\*-\"}] */\nalert('test');";
+        const code = "/* eslint max-len: [2, 100, 2, {ignoreUrls: true, ignorePattern: \"data:image\\/|\\\\s*require\\\\s*\\\\(|^\\\\s*loader\\\\.lazy|-\\\\*-\"}] */\nalert('test');";
 
         it("should not parse errors, should report a violation", () => {
             const messages = linter.verify(code, {}, filename);
 
             assert.strictEqual(messages.length, 1);
             assert.strictEqual(messages[0].ruleId, "max-len");
-            assert.strictEqual(messages[0].message, "This line has a length of 122. Maximum allowed is 100.");
+            assert.strictEqual(messages[0].message, "This line has a length of 128. Maximum allowed is 100.");
+            assert.include(messages[0].nodeType, "Program");
+        });
+    });
+
+    describe("when evaluating code with comments that contain escape sequences", () => {
+        const code = '/* eslint max-len: ["error", 1, { ignoreComments: true, ignorePattern: "console\\.log\\\\(" }] */\nconsole.log("test");\nvar a = "test2";';
```

Comment:
I don't think the `"...console\\.log..."` does anything, since this is `console\.log`, when when passed through `levn` is just `console.log`

Reply:
When you switch to using `String.raw` this will become more clear

Reply:
This is the issue this PR fixes. Since this string is evaluated at runtime and then used to generate a RegExp, we need to ensure that the `.` is escaped (so that it's evaluated as a literal `.` character in the RegExp).

Reply:
You should add a test case like `consoleXlog("test");` and make sure it does not fire if it's anything but a dot.

I'm saying is that what you actually want is an escaped backslash `\`, so you want `\\` (under `String.raw` conditions). In JS console type `"console\.log"`, you will see that the output is `console.log`. You get this right later on when you ensure there is an escaped backlash before the `(` by doing `\\(`, so that it is escaped in the regex

Reply:
OH, I understand now - thanks! Updating now.